cleverhans/future/jax/attacks/fast_gradient_method.py

```diff
@@ -0,0 +1,56 @@
+import jax.numpy as np
+from jax import grad, vmap
+from jax.experimental.stax import logsoftmax
+
+from cleverhans.future.jax.utils import one_hot
+
+
+def fast_gradient_method(model_fn, x, eps, ord, clip_min=None, clip_max=None, y=None,
+	targeted=False):
+  """
+  JAX implementation of the Fast Gradient Method.
+  :param model_fn: a callable that takes an input tensor and returns the model logits.
+  :param x: input tensor.
+  :param eps: epsilon (input variation parameter); see https://arxiv.org/abs/1412.6572.
+  :param ord: Order of the norm (mimics NumPy). Possible values: np.inf, 1 or 2.
+  :param clip_min: (optional) float. Minimum float value for adversarial example components.
+  :param clip_max: (optional) float. Maximum float value for adversarial example components.
+  :param y: (optional) Tensor with true labels. If targeted is true, then provide the
```

Comment:
Let's clarify that y is probability vector, possibly one-hot for a single class target.